### PR TITLE
add make_model_data_from_onnx_data_on_memory()

### PR DESCRIPTION
**Pull request by okdshin**
`make_model_data_from_onnx_data_on_memory()` enables to construct `model_data` from ONNX protobuf serialized binary data on memory.

### CODE REVIEW

File: menoh/menoh.cpp [hunk 1]
```diff
@@ -36,7 +36,9 @@ namespace menoh_impl {
             auto cont =
               std::copy(prefix, prefix + std::char_traits<char>::length(prefix),
                         arr.begin());
-            std::copy(message, message + (static_cast<size_t>(arr.end() - cont) - 1), cont);
+            std::copy(message,
+                      message + (static_cast<size_t>(arr.end() - cont) - 1),
+                      cont);
```
Comment (msakai) [inline menoh/menoh.cpp]: It would be nice to separate stylistic changes into a different commit, because they may blur essential changes.

File: menoh/node.hpp [hunk 2]
```diff
@@ -20,6 +20,7 @@ namespace menoh_impl {
     };
 
     bool operator==(node const& a, node const& b);
+    inline bool operator!=(node const& a, node const& b) { return !(a == b); }
```
Comment (msakai) [inline menoh/node.hpp]: Is this necessary for the objective of this PR?
If not, how about removing this from this PR and applying the change separately?

File: test/common.hpp [hunk 3]
```diff
@@ -40,8 +43,27 @@ namespace menoh_impl {
...
+        }
+        auto size = total_size(lhs);
+        for(auto i = 0; i < size; ++i) {
+            float l = *(static_cast<float*>(lhs.data()) + i);
+            float r = *(static_cast<float*>(rhs.data()) + i);
+            if(!(l - eps < r && r < l + eps)) {
+                return false;
+            }
+        }
+        return true;
```
Comment (msakai) [inline test/common.hpp]: Is this necessary for the objective of this PR?
If not, how about removing this from this PR and applying the change separately?

File: test/onnx.cpp [hunk 4]
```diff
@@ -46,5 +52,45 @@ namespace menoh_impl {
             }
         }
 
+        TEST_F(ONNXTest, make_model_data_from_onnx_data_on_memory) {
+            std::ifstream input(model_filename, std::ios::binary);
+            std::istreambuf_iterator<char> begin(input);
+            std::istreambuf_iterator<char> end;
```
Comment (msakai) [inline test/onnx.cpp]: Is the `end` variable initialized properly?
This might be trivial for C++ programmer but I'm not familiar with C++.

Comment (okdshin) [reply test/onnx.cpp]: Yes. It is initialized by its default constructor.

Comment (msakai) [reply test/onnx.cpp]: Thanks!

File: test/common.hpp (hunk 3 above)
Comment (okdshin) [reply test/common.hpp]: `is_near_array` is used for testing [here](https://github.com/pfnet-research/menoh/pull/88/files#diff-a07ff2d45ccdec4e633eb369f99677bbR91)

File: menoh/node.hpp (hunk 2 above)
Comment (okdshin) [reply menoh/node.hpp]: You are correct. It is not necessary for this PR. I will remove it.

File: test/common.hpp (hunk 3 above)
Comment (msakai) [reply test/common.hpp]: Oop! I overlooked that.
Thanks!

Comment (okdshin) [reply test/common.hpp]: I will add some comments into testing code.

**Comment by msakai** [comment]
Looks good to me.
@colspan, Do you also review?
